hack/run-openshift

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+#
+# Test the MongoDB image in OpenShift.
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+
+set -exo nounset
+
+function prepare_oc_client() {
+  # Prepare oc command
+  if [[ ! -d ../openshift-client/ ]]; then
+    OC_RELEASE=https://github.com/openshift/origin/releases/download/v1.4.1/openshift-origin-client-tools-v1.4.1-3f9807a-linux-64bit.tar.gz
+
+    curl -L ${OC_RELEASE} | tar xz
+    mv ./openshift-origin-client-tools-* ../openshift-client
+  fi
+  export PATH=${PATH}:`pwd`/../openshift-client/
+}
+prepare_oc_client
+
+function cleanup() {
+  echo "Stopping and removing OpenShift cluster..."
+  oc cluster down
+}
+trap cleanup EXIT SIGINT
+
+function print_logs() {
+  oc get all
+  while read pod_info; do
+    pod=$(echo ${pod_info} | tr -s ' ' | cut -f1 -d' ')
+    echo "INFO: printing logs for pod ${pod}"
+    oc logs ${pod}
+  done < <(oc get pods --no-headers=true)
+}
+trap print_logs ERR
+
+# Start OpenShift cluster
+oc cluster up
+
+# Prepare image - add tag different than :latest
+# With :latest default imagePullPolicy is Always
+# https://docs.openshift.com/enterprise/3.2/dev_guide/managing_images.html#image-pull-policy
+docker tag ${IMAGE_NAME} ${IMAGE_NAME}:test-openshift
+IMAGE_NAME=${IMAGE_NAME}:test-openshift
+
+#
+# General functions
+#
+
+# wait_for_ready_pods wait till number of pods with label get ready
+function wait_for_ready_pods() {
+  local label="$1"
+  local count="$2"
+  for i in $(seq 30); do
+    if [[ "$(oc get pods --no-headers=true -l${label} | grep "1/1" | wc -l)" -eq ${count} ]]; then
+      return 0
+    fi
+    echo "Waiting for ${count} ready pods labeled with '${label}'..."
+    sleep 20
+  done
+  return 1
+}
+
+# create_volumes reates Kubernetes volumes with given name prefix, specified capacity and accessMode
+function create_volumes() {
```

Comment:
oc cluster up, as of 1.5, actually creates volumes automatically, so if you move to the 1.5 alpha release you can drop this.